extensions/schema-compare/src/test/schemaCompare.test.ts

```diff
@@ -85,24 +93,305 @@ describe('SchemaCompareMainWindow.start', function (): void {
 		should.equal(result.targetEndpointInfo, undefined);
 	});
 });
+let showErrorMessageSpy: any;
+let showWarningMessageStub: any;
+let showOpenDialogStub: any;
+
+describe('SchemaCompareMainWindow.results', function (): void {
+	before(() => {
+		mockExtensionContext = TypeMoq.Mock.ofType<vscode.ExtensionContext>();
+		mockExtensionContext.setup(x => x.extensionPath).returns(() => '');
+	});
+
+	this.afterEach(() => {
+		sinon.restore();
+	});
+
+	this.beforeEach(() => {
+		sinon.restore();
+		showErrorMessageSpy = sinon.spy(vscode.window, 'showErrorMessage');
+	});
+
+	it('Should show error if publish changes fails', async function (): Promise<void> {
+		let service = createServiceMock();
+		service.setup(x => x.schemaComparePublishChanges(TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => Promise.resolve({
+			success: false,
+			errorMessage: ''
+		}));
+
+		showWarningMessageStub = sinon.stub(vscode.window, 'showWarningMessage').returns(<any>Promise.resolve('Yes'));
+
+		let schemaCompareResult = new SchemaCompareMainWindow(service.object, mockExtensionContext.object);
+		await schemaCompareResult.start(undefined);
+
+		schemaCompareResult.sourceEndpointInfo = setDacpacEndpointInfo(mocksource);
+		schemaCompareResult.targetEndpointInfo = setDacpacEndpointInfo(mocktarget);
+		await schemaCompareResult.execute();
+		await schemaCompareResult.publishChanges();
+
+		should(showErrorMessageSpy.calledOnce).be.true();
```

Comment:
can there also be a check that the error message matches the expected text? Also for the other tests that are checking that an error message gets shown